datasets/wili_2018/README.md

```diff
@@ -0,0 +1,155 @@
+---
+annotations_creators:
+- no-annotation
+language_creators:
+- found
+languages:
+- 235 different languages
+licenses:
+- ODC Open Database License v1.0
+multilinguality:
+- 235 different languages
+size_categories:
+- 1000 lines of text for every language, so in total 235000 lines of text (as total 235 languages)
+source_datasets:
+- original
+task_categories:
+- language-identification
+task_ids:
+- language-identification
+---
+
+# Dataset Card for wili_2018
+
+## Table of Contents
+- [Dataset Description](#dataset-description)
+  - [Dataset Summary](#dataset-summary)
+  - [Supported Tasks](#supported-tasks-and-leaderboards)
+  - [Languages](#languages)
+- [Dataset Structure](#dataset-structure)
+  - [Data Instances](#data-instances)
+  - [Data Fields](#data-instances)
+  - [Data Splits](#data-instances)
+- [Dataset Creation](#dataset-creation)
+  - [Curation Rationale](#curation-rationale)
+  - [Source Data](#source-data)
+  - [Annotations](#annotations)
+  - [Personal and Sensitive Information](#personal-and-sensitive-information)
+- [Considerations for Using the Data](#considerations-for-using-the-data)
+  - [Social Impact of Dataset](#social-impact-of-dataset)
+  - [Discussion of Biases](#discussion-of-biases)
+  - [Other Known Limitations](#other-known-limitations)
+- [Additional Information](#additional-information)
+  - [Dataset Curators](#dataset-curators)
+  - [Licensing Information](#licensing-information)
+  - [Citation Information](#citation-information)
+
+## Dataset Description
+
+- **Homepage:** https://zenodo.org/record/841984
+- **Repository:** [Needs More Information]
+- **Paper:** https://arxiv.org/pdf/1801.07779
+- **Leaderboard:** [Needs More Information]
+- **Point of Contact:** Thoma, Martin (Email: info@martin-thoma.de)
+
+### Dataset Summary
+
+WiLI-2018, the Wikipedia language identification benchmark dataset, contains 235000 paragraphs of 235 languages. The dataset is balanced and a train-test split is provided.
+
+### Supported Tasks and Leaderboards
+
+[Needs More Information]
+
+### Languages
+
+235 Different Languages
+
+## Dataset Structure
+
+### Data Instances
+
+* `x_train.txt`: 175000 lines of text. Each line belongs to one language.
+* `y_train.txt`: 175000 lines. Each line denotes the language of the same line
+                 in `x_train.txt`
+* `x_test.txt`: See `x_train.txt`
+* `y_test.txt`: See `y_train.txt`
+* `urls.txt`: A list of permanent URLs to all pages used for paragraph extraction
+* `labels.csv`: A header line plus one line per language
+* `README.txt`: This file
```

Comment:
One of the sample example, which I got is:

{'label': 207, 'sentence': 'Ti Turkia ket maysa a demokrata, sekular, unitario, batay-linteg a republika nga addaan ti taga-ugma a tinawtawid a kultura. Ti Turkia ket umadadu a naipatipon iti Laud babaen ti panagkameng kadagiti organisasion a kas ti Konsilo iti Europa, NATO, OECD, OSCE ken ti G-20 a dagiti kangrunaan nga ekonomia. Ti Turkia ket nangrugi a nakitulag ti napno a panagkameng iti Kappon ti Europa idi 2005, nga isu ket maysa idin a kumaduaan a kameng iti Europeano a Komunidad ti Ekonomia manipud idi 1963 ken nakadanon ti maysa a tulagan ti kappon ti aduana idi 1995. Ti Turkia ket nagtaraken iti asideg a kultural, politikal, ekonomiko ken industria a panakibiang iti Tengnga a Daya, dagiti Turko nga estado iti Tengnga nga Asia ken dagiti pagilian ti Aprika babaen ti panagkameng kadagiti organisasion a kas ti Turko a Konsilo, Nagsaupan nga Administrasion iti Turko nga Arte ken Kultura, Organisasion iti Islamiko a Panagtitinnulong ken ti Organisasion ti Ekonomiko a Panagtitinnulong.'}

Reply:
Okay, will copy-paste that.

Reply:
And I tried to use CSV reader instead of pandas but wasn't able to succeed in it. 
For example, a data observation looks like this (one observation in each line):
Hi, I am a human, 10
^                       ^^ ^
|................................| |   |
     sentence         ,   label

So if I try separate sentences and labels with a delimiter ',' (and each line denotes one example), I wasn't able to do it as a sentence may include a ',' in it. @lhoestq Can you suggest something else, that I can try?

Reply:
You simply do something like
```python
with open(filepath, encoding="utf-8") as f:
    for id_, line in enumerate(f):
        split_index = line.rstrip(",")
        text = line[:split_index].strip('"')
        label = int(lines[split_index + 1:].strip())
        yield id_, {...}
```